pilot/pkg/networking/plugin/plugin.go

```diff
@@ -51,10 +52,22 @@ const (
 )
 
 // ModelProtocolToListenerProtocol converts from a config.Protocol to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(node *model.Proxy, p protocol.Instance) ListenerProtocol {
+func ModelProtocolToListenerProtocol(node *model.Proxy, p protocol.Instance,
+	trafficDirection core.TrafficDirection) ListenerProtocol {
 	// If protocol sniffing is not enabled, the default value is TCP
-	if !util.IsProtocolSniffingEnabledForNode(node) && p == protocol.Unsupported {
-		p = protocol.TCP
+	if p == protocol.Unsupported {
+		switch trafficDirection {
+		case core.TrafficDirection_INBOUND:
+			if !util.IsProtocolSniffingEnabledForInbound(node) {
+				p = protocol.TCP
+			}
+		case core.TrafficDirection_OUTBOUND:
+			if !util.IsProtocolSniffingEnabledForNode(node) {
```

Comment:
dont you want to do a search and replace and call this IsProtocolSniffingEnabledForOutbound ?

Reply:
Renamed